src/common/tse.c

```diff
@@ -642,12 +644,19 @@ tse_task_post_process(tse_task_t *task)
 		if (task_tmp->dt_result == 0 && !dtp_tmp->dtp_no_propagate)
 			task_tmp->dt_result = task->dt_result;
 
+		dsp_tmp = dtp_tmp->dtp_sched;
+		diff_sched = dsp != dsp_tmp;
+
+		if (diff_sched) {
+			D_MUTEX_UNLOCK(&dsp->dsp_lock);
+			D_MUTEX_LOCK(&dsp_tmp->dsp_lock);
```

Comment:
just curious, why unlock dsp here? to avoid out of order deadlock?

Reply:
right, I think cannot try to hold two different scheduler's lock, that possibly cause dead lock.